platform/security/idp/security-idp-server/src/main/java/org/codice/ddf/security/idp/server/IdpEndpoint.java

```diff
@@ -0,0 +1,820 @@
+                        }
+                    };
+                    signatureCrypto = CryptoFactory
+                            .getInstance(sigProperties, IdpEndpoint.class.getClassLoader(),
+                                    passwordEncryptor);
+                } catch (WSSecurityException e) {
+                    LOGGER.debug("Error in loading the signature Crypto object: ", e);
+                }
+            } else {
+                LOGGER.debug("Cannot load signature properties using: " + signaturePropertiesPath);
+            }
+        }
+        return signatureCrypto;
+    }
+
+    protected Crypto getEncryptionCrypto() {
+        if (encryptionCrypto == null && encryptionPropertiesPath != null) {
+            Properties sigProperties = PropertiesLoader.loadProperties(encryptionPropertiesPath);
+            if (sigProperties != null) {
+                try {
+                    PasswordEncryptor passwordEncryptor = new PasswordEncryptor() {
+                        @Override
+                        public String encrypt(String password) {
+                            return encryptionService.encrypt(password);
+                        }
+
+                        @Override
+                        public String decrypt(String encryptedPassword) {
+                            return encryptionService.decrypt(encryptedPassword);
+                        }
+                    };
+                    encryptionCrypto = CryptoFactory
+                            .getInstance(sigProperties, IdpEndpoint.class.getClassLoader(),
+                                    passwordEncryptor);
+                } catch (WSSecurityException e) {
+                    LOGGER.debug("Error in loading the signature Crypto object: ", e);
+                }
+            } else {
+                LOGGER.debug("Cannot load signature properties using: " + encryptionPropertiesPath);
+            }
+        }
+        return encryptionCrypto;
+    }
+
+    protected String getSignaturePassword() {
+        if (signaturePropertiesPath != null) {
+            Properties sigProperties = PropertiesLoader.loadProperties(signaturePropertiesPath);
+            if (sigProperties != null) {
+                String password = sigProperties
+                        .getProperty(Merlin.PREFIX + Merlin.KEYSTORE_PRIVATE_PASSWORD);
+                if (password == null) {
+                    password = sigProperties
+                            .getProperty(Merlin.OLD_PREFIX + Merlin.KEYSTORE_PRIVATE_PASSWORD);
+                }
+                if (password != null) {
+                    password = password.trim();
+                    password = decryptPassword(password, encryptionService);
+                }
+                return password;
+            }
+        }
+        return null;
+    }
+
+    protected String getEncryptionPassword() {
+        if (encryptionPropertiesPath != null) {
```

Comment:
The only difference between this method and `getSignaturePassword()` is the properties path. Could easily refactor.